tools/infrasys/src/main.rs

```diff
@@ -0,0 +1,268 @@
+mod error;
+mod keys;
+mod root;
+mod s3;
+mod shared;
+
+use error::Result;
+use log::{error, info};
+use pubsys_config::InfraConfig;
+use sha2::{Digest, Sha512};
+use shared::KeyRole;
+use simplelog::{Config as LogConfig, LevelFilter, SimpleLogger};
+use snafu::{ensure, OptionExt, ResultExt};
+use std::path::{Path, PathBuf};
+use std::{fs, process};
+use structopt::StructOpt;
+use tokio::runtime::Runtime;
+use url::Url;
+
+//   =^..^=   =^..^=   =^..^=  SUB-COMMAND STRUCTS  =^..^=   =^..^=   =^..^=
+
+#[derive(Debug, StructOpt)]
+#[structopt(setting = clap::AppSettings::DeriveDisplayOrder)]
+struct Args {
+    #[structopt(global = true, long, default_value = "INFO")]
+    log_level: LevelFilter,
+
+    // Path to Infra.toml  (NOTE: must be specified before subcommand)
+    #[structopt(long, parse(from_os_str))]
+    infra_config_path: PathBuf,
+
+    #[structopt(subcommand)]
+    subcommand: SubCommand,
+}
+
+#[derive(Debug, StructOpt)]
+#[structopt(setting = clap::AppSettings::DeriveDisplayOrder)]
+struct CreateInfraArgs {
+    #[structopt(long)]
+    root_role_path: PathBuf,
+}
+
+#[derive(Debug, StructOpt)]
+enum SubCommand {
+    CheckInfraLock,
+    CreateInfra(CreateInfraArgs),
```

Comment:
These doc comments become help comments when you run `infrasys --help`

```suggestion
    /// Creates infrastructure specified in the Infra.toml file.
    CreateInfra(CreateInfraArgs),
```